x-pack/plugins/task_manager/server/create_task_manager.ts

```diff
@@ -27,19 +26,16 @@ export function createTaskManager(
   {
     logger,
     config,
-    savedObjectSchemas,
     elasticsearch: { callAsInternalUser },
     savedObjectsRepository,
+    savedObjectsSerializer,
   }: LegacyDeps
 ) {
-  // as we use this Schema solely to interact with Tasks, we
-  // can initialise it with solely the Tasks schema
-  const serializer = new SavedObjectsSerializer(new SavedObjectsSchema(savedObjectSchemas));
   return new TaskManager({
     taskManagerId: core.uuid.getInstanceUuid(),
     config,
     savedObjectsRepository,
-    serializer,
+    serializer: savedObjectsSerializer,
```

Comment:
@elastic/kibana-alerting-services the serializer now requires an internal core type to be instantiated. I exposed an API from core to retrieve a configured serializer, and adapted your call to use it instead.

The `createTaskManager` method no longer relies on any legacy API, however I did not feel confident on impacting too much you codebase, so I kept the call as is.

You should be good to remove the `registerLegacyAPI` trick in `x-pack/legacy/plugins/task_manager/server/index.ts` and  `x-pack/plugins/task_manager/server/plugin.ts` once this lands.